src/runtime/virtcontainers/hypervisor.go

```diff
@@ -894,6 +909,16 @@ func genericAvailableGuestProtections() (protections []string) {
 	return
 }
 
+func GetHugepageTypes() []string {
+	arch := runtime.GOARCH
+	if arch == "amd64" {
+		return []string{HugepageSize2M, HugepageSize1G}
+	} else if arch == "arm64" {
+		return []string{HugepageSize2M, HugepageSize512M, HugepageSize1G}
+	}
+	return nil
+}
+
```

Comment:
All `GetHugepageType()` invocations so far follow this pattern:

```
	hugepageTypes := GetHugepageTypes()
	for _, t := range hugepageTypes {
		if t == valueBeingChecked { do something }
	}
```
To reduce the amount of lowlevel iteration and branching code, how about adding something like
```
func isHugepageSizeValid(sz string) bool {
	hugepageTypes := GetHugepageTypes()
	for _, t := range hugepageTypes {
		if t == sz { return true }
	}
	return false
}
```
(Please don't mind the formatting, it's kind of tedious to write code in a comment. :-))